### DISCUSSION TITLE
Tasmota 10 - flash writes

### DISCUSSION THREAD

**Discussion by bgolab** [arendst/Tasmota#14341] [Q&A] [answered]
Hi,
I am trying to understand all cases when the Tasmota writes something to the flash and how to disable flash writes.
1. Power state - controlled by SetOption0
2. Parameter Changes - controlled  by SaveData
3. Energy Sensor - the energy data are still saved regardless if the SaveData is set to 0 or not - is there any way how to prevent saving anything here
4. Web GUI restart - i t looks like the SaveCount increased after such action like Web GUI restart

Any comments?

**Comment by barbudor** [comment, marked as answer]
3 should only occur once a day at midnight
4 should occur only if there is something to save

**Reply by bgolab** [reply]
Ad 3, yes agree
Ad 4. I did several subsequent Web GUI restarts and I believe every time the the SaveCount increases. 
At the beginning I assumed that the Web GUI restart is the equivalent to 'restart 1' but I am not sure now.
My goal was to have Tasmota setup in the state where nothing is saved without explicit request.

I monitor flash writes through a simple script (curl/jq employed) parsing json data - this matches what I see from the WebGUI.

**Reply by sfromis** [reply]
Docs for the [restart](https://tasmota.github.io/docs/Commands/#restart) command (which is the same as doing it from the web UI) explicitly states *restart device with configuration saved to flash*. It also has *99 = force restart device without configuration save*

For context, part of the data saved is boot count, thus there is an update for every boot.

With deep sleep, you also have [SetOption76](https://tasmota.github.io/docs/Commands/#setoption76) to control whether boot count is updated, to mitigate the many flash writes which otherwise would be a part of having a sensor device wake up to report data at regular intervals.

**Reply by bgolab** [reply]
Yes, I was wondering why the save is enforced here when I does not change anything. 
But the boot count update explains the behavior of the Web restart (means there is a change).

Thank you.
